src/builtins.ts

```diff
@@ -527,8 +530,77 @@ function isinstance(t: Thread, f: IPy_FrameObj, args: IPy_Object[], kwargs: Py_D
   }
 }
 
+//Exception classes
+class BaseException extends Py_Object {
+  $__dict__ = new Py_Dict(<any> this);
+  $args: Py_Tuple;
+  $message: Py_Str;
+
+  $__mro__: Py_Tuple;
+  $__call__: Py_SyncNativeFuncObject;
+  $__getstate__: Py_SyncNativeFuncObject;
+  $__setstate__: Py_SyncNativeFuncObject;
+
+  constructor(args?: IPy_Object[]) {
+    super();
+    if (args) {
+      this.$args = new Py_Tuple(args);
+    }
+  }
+}
+
+BaseException.prototype.$__mro__ = new Py_Tuple([BaseException.prototype, Py_Object.prototype]);
+BaseException.prototype.$__call__ = new Py_SyncNativeFuncObject((t: Thread, f: interfaces.IPy_FrameObj, args: IPy_Object[], kwargs: Py_Dict) => {
+  return new BaseException(args);
+});
+BaseException.prototype.$__getstate__ = new Py_SyncNativeFuncObject((t: Thread, f: interfaces.IPy_FrameObj, args: IPy_Object[], kwargs: Py_Dict) => {
+  return this.$args;
+});
+BaseException.prototype.$__setstate__ = new Py_SyncNativeFuncObject((t: Thread, f: interfaces.IPy_FrameObj, args: IPy_Object[], kwargs: Py_Dict) => {
+  this.$__dict__ = kwargs;
+  return kwargs;
+});
+
+
+// TODO: Change exception classes so that they print $args during tracebacks for user-defined exc classes
+class Exception extends BaseException {
+  constructor(args?: IPy_Object[]){
```

Comment:
Same comment here as with `NameError`. Doesn't `BaseException`'s constructor do this already?